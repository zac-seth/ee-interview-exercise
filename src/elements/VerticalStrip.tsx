import { FC } from 'react';

import { LayoutDirection } from './layout-direction';
import { Layout } from './Layout.style';

export type VerticalStripProps = { length: number };

export const VerticalStrip: FC<VerticalStripProps> = ({ children, length }) => (
  <Layout
    layoutDirection={LayoutDirection.COLUMN}
    length={length}
    data-testid='vertical-strip-layout'
  >
    {children}
  </Layout>
);
