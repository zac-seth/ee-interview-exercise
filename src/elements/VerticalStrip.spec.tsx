import { render, screen } from '@testing-library/react';

import { VerticalStrip } from './VerticalStrip';

const LENGTH = 3;

const makeChildren = (length: number): JSX.Element[] =>
  Array.from({ length }, (_v, i) => (
    <div data-testid={`item-${i}`} key={`key-${i}`}>
      value {i}
    </div>
  ));

const makeStrip = () => (
  <VerticalStrip length={LENGTH}>{makeChildren(3)}</VerticalStrip>
);

describe(`VerticalStrip`, () => {
  it(`Should render the VerticalStrip with the list of given child items`, () => {
    render(makeStrip());

    expect(screen.getByTestId(`vertical-strip-layout`)).toMatchSnapshot();
  });

  it(`Should render the grid-template-rows direction attribute with the expected value`, () => {
    render(makeStrip());

    const element = screen.getByTestId(`vertical-strip-layout`);

    expect(element).toHaveStyleRule(`grid-template-rows`, `repeat(3,1fr)`);
  });

  it(`Should render the children in the VerticalStrip`, async () => {
    render(makeStrip());

    const children = await screen.findAllByTestId(`item`, { exact: false });

    expect(children.length).toBe(3);
  });
});
