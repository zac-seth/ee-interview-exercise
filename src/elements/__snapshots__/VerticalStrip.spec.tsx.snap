// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerticalStrip Should render the VerticalStrip with the list of given child items 1`] = `
.c0 {
  display: grid;
  grid-gap: 0.25rem;
  margin: 0;
  padding: 0;
  grid-template-rows: repeat(3,1fr);
}

<div
  class="c0"
  data-testid="vertical-strip-layout"
>
  <div
    data-testid="item-0"
  >
    value 
    0
  </div>
  <div
    data-testid="item-1"
  >
    value 
    1
  </div>
  <div
    data-testid="item-2"
  >
    value 
    2
  </div>
</div>
`;
